Server/MirObjects/Monsters/AncientBringer.cs

```diff
@@ -0,0 +1,210 @@
+                FindTarget();
+
+        }
+
+        private void LineAttack1(int distance)
+        {
+            int damage = GetAttackPower(MinDC, MaxDC);
+            if (damage == 0) return;
+            Target.Attacked(this, damage, DefenceType.ACAgility);
+        }
+
+        Cell cell = null;
+
+        private void LineAttack2(int distance)
+        {
+            MirDirection dir = Functions.DirectionFromPoint(Target.CurrentLocation, CurrentLocation);
+            dir = Functions.NextDir(dir);
+            Point target = Functions.PointMove(CurrentLocation, dir, 1);
+
+            int damage = GetAttackPower(MinDC, MaxDC * 2);
+            if (damage == 0) return;
+            Target.Attacked(this, damage, DefenceType.ACAgility);
+
+            for (int i = 0; i < 4; i++)
+            {
+                target = Functions.PointMove(CurrentLocation, dir, 1);
+                dir = Functions.NextDir(dir);
+
+                if (!CurrentMap.ValidPoint(target)) continue;
+
+                cell = CurrentMap.GetCell(target);
+
+                if (cell.Objects == null) continue;
+
+                for (int o = 0; o < cell.Objects.Count; o++)
+                {
+                    MapObject ob = cell.Objects[o];
+                    if (ob.Race != ObjectType.Player && ob.Race != ObjectType.Monster) continue;
+                    if (!ob.IsAttackTarget(this)) continue;
+
+                    ob.Attacked(this, damage, DefenceType.ACAgility);
+
+                    if (Envir.Random.Next(Settings.PoisonResistWeight) >= Target.PoisonResist)
+                    {
+                        if (Envir.Random.Next(5) == 0)
+                        {
+                            Target.ApplyPoison(new Poison { Owner = this, Duration = 5, PType = PoisonType.Paralysis, Value = GetAttackPower(MinSC, MaxSC), TickSpeed = 2000 }, this);
+                        }
+                    }
+
+                    break;
+                }
+            }
+        }
+
+        private void RangeAttack1(int distance)
+        {
+            int damage = GetAttackPower(MinMC, MaxMC);
+            if (damage == 0) return;
+            DelayedAction action = new DelayedAction(DelayedType.Damage, Envir.Time + 300, Target, damage, DefenceType.MAC);
+            Target.Attacked(this, damage, DefenceType.MACAgility);
+        }
+
+        private void RangeAttack2(int distance)
+        {
+            int damage = GetAttackPower(MinMC, MaxMC * 2);
+            if (damage == 0) return;
+            Target.Attacked(this, damage, DefenceType.MACAgility);
+
+            SpawnSlaves();
+        }
+
+        private void SpawnSlaves()
+        {
+            int count = Math.Min(6, 40 - SlaveList.Count);
+
+            ActionTime = Envir.Time + 300;
+            AttackTime = Envir.Time + AttackSpeed;
+
+            for (int i = 0; i < count; i++)
+            {
+                MonsterObject mob = null;
+                switch (Envir.Random.Next(4))
```

Comment:
whats the point of this switch? will random between 0 and 3, but all of the outcomes give you the same mob?